src/OpenTelemetry/Internal/WildcardHelper.cs

```diff
@@ -41,7 +41,17 @@ public static Regex GetWildcardRegex(IEnumerable<string> patterns = default)
 
         var convertedPattern = string.Join(
             "|",
-            from p in patterns select "(?:" + Regex.Escape(p).Replace("\\*", ".*").Replace("\\?", ".") + ')');
+            from p in patterns select "(?:" + ReplacePatterns(p) + ')');
         return new Regex('^' + convertedPattern + '$', RegexOptions.Compiled | RegexOptions.IgnoreCase);
```

Comment:
I think the fix should be this one line change:

```suggestion
        return new Regex("^(" + convertedPattern + ")$", RegexOptions.Compiled | RegexOptions.IgnoreCase);
```

Reply:
Or maybe:

```suggestion
        return new Regex("^(?:" + convertedPattern + ")$", RegexOptions.Compiled | RegexOptions.IgnoreCase);
```

@Yun-Ting what's your suggestion?

Reply:
Non-capturing group should be sufficient here.